### PR TITLE
DOCS-#2436: Explicit local / single node backend

### LINKED ISSUE #2436: Explicit backend use in case of LOCAL run
When we run MODIN in local whitout providing any dask or .. cluster does it run with the [multiprocessing ](https://docs.python.org/fr/3/library/multiprocessing.html) package of the lib python?  This could be explicited in the documentation and README.md

At first I was looking how and why would modin run  a "fake" dask cluster in local.

What do you think about it ? Thank you

### PR DESCRIPTION
**Pull request by raphaelauv**
Explain how in local ( single node ) modin work and use a simulated dark or ray cluster

Closes #2436

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,6 +131,8 @@ import numpy as np
 frame_data = np.random.randint(0, 100, size=(2**10, 2**8))
 df = pd.DataFrame(frame_data)
 ```
+**In local (without a cluster) modin will create and manage a simulated (dask or ray) cluster for the execution**
```
Comment (devin-petersohn) [inline README.md:134]: Can we change the word `simulated` to `local`? Simulated is not a term that will be familiar to the readers.

File: docs/using_modin.rst [hunk 2]
```diff
@@ -21,6 +21,8 @@ functional parity with pandas.
 Using Modin on a Single Node
 ----------------------------
 
+**In local (without a cluster) modin will create and manage a simulated (dask or ray) cluster for the execution**
```
Comment (devin-petersohn) [inline docs/using_modin.rst:24]: Same here for `simulated` -> `local`

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
Thanks @raphaelauv, great addition!

One wording change for using terminology more common to users.

**Comment by devin-petersohn** [comment]
@raphaelauv Thanks for the patience with the review, it was a holiday last week in the US, so many were not able to review it yet.

File: README.md (hunk 1 above)
Comment (raphaelauv) [reply README.md:134]: Okay

**Comment by raphaelauv** [comment]
@devin-petersohn thank for your review , I updated the PR

**Review by anmyachev — APPROVED** [review]
Thanks @raphaelauv!

**Review by devin-petersohn — APPROVED** [review]

**Pull request merged**